### ISSUE TITLE
Wrong doc: Spec.{before,after}_each cannot be run anywhere

### ISSUE THREAD

**Issue by bew** [crystal-lang/crystal#4304]
From the docs, [Spec.before_each](https://crystal-lang.org/api/0.21.1/Spec.html#before_each(&block)-class-method) & [Spec.after_each](https://crystal-lang.org/api/0.21.1/Spec.html#after_each(&block)-class-method) says that it works `regardless of where this method is invoked`.

But that's currently not true:
```cr
require "spec"

describe "bla" do
  # calls are before
  Spec.before_each do
    puts "before"
  end
  
  Spec.after_each do
    puts "after"
  end
  
  it "works" do
    true.should be_true
  end
end
```
Gives: https://carc.in/#/r/1vmi
```
before
.after


Finished in 87 microseconds
1 examples, 0 failures, 0 errors, 0 pending
```
But if I put the calls after:
```cr
require "spec"

describe "bla" do
  it "works" do
    true.should be_true
  end
  
  # calls are after
  Spec.before_each do
    puts "before"
  end
  
  Spec.after_each do
    puts "after"
  end
end
```
Gives: https://carc.in/#/r/1vmj
```
.

Finished in 64 microseconds
1 examples, 0 failures, 0 errors, 0 pending
```
Did I miss something?

**Comment by asterite** [comment]
No, they must be specified before the specs. You generally put them in a `spec_helper` of some sort. They are applied globally to every spec that comes after they are invoked.

**Comment by asterite** [comment]
The note "regardless of where they are invoked" should really say that they apply globally to every spec that comes after them.

**Comment by bew** [comment]
Oh I see, maybe it should be rewritten to better reflect what you said?

**Comment by asterite** [comment]
I'd like to rethink spec to support setup and teardown, maybe it will change more than that so I'd wait for now.

**Comment by bew** [comment]
@asterite should this issue be closed or do you want to keep it open for future follow-up?

**Comment by jhass** [comment]
Let's document the behavior for now.

**Comment by bew** [comment]
In latest crystal (0.32.1), the behavior is consistent, the `before` & `after` always show up!

before it block: https://carc.in/#/r/8ijt
after it block: https://carc.in/#/r/8iju

**Comment by straight-shoota** [comment]
This was likely fixed by the spec refactor in #8125
